src/test/java/org/opensearch/integTest/RAGToolIT.java

```diff
@@ -0,0 +1,498 @@
+        Exception exception = assertThrows(ResponseException.class, () -> executeAgent(agentId, "{\"parameters\": {\"question\": \"a\"}}"));
+
+        org.hamcrest.MatcherAssert
+            .assertThat(
+                exception.getMessage(),
+                allOf(containsString("no such index [test_index2]"), containsString("IndexNotFoundException"))
+            );
+    }
+
+    public void testRAGToolWithNeuralSparseQuery_withIllegalModelIdField_thenThrowException() {
+        String agentId = createAgent(registerAgentWithNeuralSparseQueryRequestBody.replace(sparseEncodingModelId, "test_model_id"));
+        Exception exception = assertThrows(ResponseException.class, () -> executeAgent(agentId, "{\"parameters\": {\"question\": \"a\"}}"));
+
+        org.hamcrest.MatcherAssert
+            .assertThat(exception.getMessage(), allOf(containsString("Failed to find model"), containsString("OpenSearchStatusException")));
+    }
+
+    public void testRAGToolWithNeuralSparseQueryAndLLM_withIllegalModelIdField_thenThrowException() {
+        String agentId = createAgent(registerAgentWithNeuralSparseQueryAndLLMRequestBody.replace(sparseEncodingModelId, "test_model_id"));
+        Exception exception = assertThrows(ResponseException.class, () -> executeAgent(agentId, "{\"parameters\": {\"question\": \"a\"}}"));
+
+        org.hamcrest.MatcherAssert
+            .assertThat(exception.getMessage(), allOf(containsString("Failed to find model"), containsString("OpenSearchStatusException")));
+    }
+
+    public void testRAGToolWithNeuralQuery_withIllegalModelIdField_thenThrowException() {
+        String agentId = createAgent(registerAgentWithNeuralQueryRequestBody.replace(textEmbeddingModelId, "test_model_id"));
+        Exception exception = assertThrows(ResponseException.class, () -> executeAgent(agentId, "{\"parameters\": {\"question\": \"a\"}}"));
+
+        org.hamcrest.MatcherAssert
+            .assertThat(exception.getMessage(), allOf(containsString("Failed to find model"), containsString("OpenSearchStatusException")));
+    }
+
+    public void testRAGToolWithNeuralQueryAndLLM_withIllegalModelIdField_thenThrowException() {
+        String agentId = createAgent(registerAgentWithNeuralQueryAndLLMRequestBody.replace(textEmbeddingModelId, "test_model_id"));
+        Exception exception = assertThrows(ResponseException.class, () -> executeAgent(agentId, "{\"parameters\": {\"question\": \"a\"}}"));
+
+        org.hamcrest.MatcherAssert
+            .assertThat(exception.getMessage(), allOf(containsString("Failed to find model"), containsString("OpenSearchStatusException")));
+    }
+
+    @Override
+    List<PromptHandler> promptHandlers() {
+        PromptHandler RAGToolHandler = new PromptHandler() {
+            @Override
+            String response(String prompt) {
+                if (prompt.contains("RAGTool")) {
```

Comment:
Can we add some lines to verify the full prompt after concatenation?

Reply:
added in  [add match full prompt](https://github.com/opensearch-project/skills/pull/177/commits/ed4f8675319c2e1984048ed78a1d48a59839a14a)